#-*-python-*-

# this is a test in which certain subjob fails to submit because of the backend exception


from GangaTest.Framework.utils import assert_cannot_submit

#the second subjob fails
j = Job()
j.backend = TestSubmitter()
j.splitter = TestSplitter()
j.splitter.backs = 3*[TestSubmitter()]
j.splitter.backs[1].fail = 'submit'
j.submit()
assert j.subjobs[0].status != 'new'
assert j.subjobs[1].status == 'new'
assert j.subjobs[2].status == 'new'
assert(len(j.subjobs)==3)

#the first subjob fails (this used to have different behaviour in Ganga4)
j = j.copy()
j.splitter.backs = 3*[TestSubmitter()]
j.splitter.backs[0].fail = 'submit'
assert_cannot_submit(j)
assert(len(j.subjobs)==0)
